### PR TITLE
enh: Refine OpenMP Test to Confirm Thread Load Balancing

### PR DESCRIPTION
**Pull request by adit4443ya**
Thread IDs are now shown to confirm active threads and ensure even distribution of iterations. The output will display "xyz" printed 10 times, with each thread ID appearing twice, indicating that the 10 iterations are evenly distributed among 5 threads.
```shell
(lf2) aditya-trivedi@Observer:~/thats_me/Main_Repo$ ./lfortran/src/bin/lfortran --openmp --link-with-gcc ./lfortran/integration_tests/openmp_01.f90 -o a --openmp-lib-dir=${CONDA_PREFIX}/lib && ./a
0
xyz
2
xyz
2
3
xyz
3
xyz
1
xyz
1
xyz
4
xyz
xyz
0
xyz
4
xyz
n =     10
Done for n =     10
```

### CODE REVIEW

**Comment by certik** [comment]
Make this a separate test. The original test is testing basic usage, this one is testing slightly more advanced usage and we want to test both.

**Comment by adit4443ya** [comment]
Done

File: integration_tests/openmp_34.f90 [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    print *, "xyz"
+end do
+!$omp end do
+!$omp end parallel
+
+print *, "n = ", n
+if (n /= 10) error stop
+end subroutine
+
+program openmp_01
```
Comment (Pranavchiku) [inline integration_tests/openmp_34.f90:21]:
```suggestion
program openmp_34
```

We follow filename as program name

**Review by Pranavchiku — DISMISSED** [review]
Register test in `integration_tests/CMakeLists.txt` file, simply apply the following diff

```diff
diff --git a/integration_tests/CMakeLists.txt b/integration_tests/CMakeLists.txt
index 340367e8c..c6c0f394c 100644
--- a/integration_tests/CMakeLists.txt
+++ b/integration_tests/CMakeLists.txt
@@ -1772,6 +1772,8 @@ RUN(NAME openmp_30 LABELS gfortran llvm_omp GFORTRAN_ARGS -fopenmp) # matmul
 RUN(NAME openmp_31 LABELS gfortran llvm_omp GFORTRAN_ARGS -fopenmp)
 RUN(NAME openmp_32 LABELS gfortran llvm_omp GFORTRAN_ARGS -fopenmp)
 RUN(NAME openmp_33 LABELS gfortran llvm_omp GFORTRAN_ARGS -fopenmp) # mandelbrot
+RUN(NAME openmp_34 LABELS gfortran llvm_omp GFORTRAN_ARGS -fopenmp)
+
 
 RUN(NAME compiler_version_01 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc llvm17)
 
```

and then run the following command to see if it works:

```console
% ./integration_tests/run_tests.py -b llvm_omp
```

File: integration_tests/openmp_34.f90 (hunk 1 above)
Comment (adit4443ya) [reply integration_tests/openmp_34.f90]: done

**Review by certik — APPROVED** [review]
Thanks!

**Pull request merged**